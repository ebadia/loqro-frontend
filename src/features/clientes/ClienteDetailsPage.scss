@import '../../styles/mixins';

.clientes-cliente-details-page {

}
