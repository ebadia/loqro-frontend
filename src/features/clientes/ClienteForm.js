import React from 'react';
import { Field } from 'react-final-form';
import { Button, Segment, Divider } from 'semantic-ui-react';
import { TextFieldAdapter } from '../../common/FormUtils';

const ClienteForm = ({ handleSubmit, submitting, pristine }) => (
  <Segment>
    <form onSubmit={handleSubmit} className="ui form clientes-cliente-form">
      <Field fluid component={TextFieldAdapter} name="nombre" label="Nombre" />
      <Divider />
      <Button
        className="clientes-cliente-form-submit"
        type="submit"
        disabled={pristine || submitting}
      >
        Enviar
      </Button>
    </form>
  </Segment>
);

export default ClienteForm;
