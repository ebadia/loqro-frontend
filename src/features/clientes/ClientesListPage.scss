@import '../../styles/mixins';

.clientes-clientes-list-page {

}
