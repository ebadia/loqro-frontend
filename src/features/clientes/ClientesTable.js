import React from 'react';
import PropTypes from 'prop-types';
import { Icon, Segment, Table, Button } from 'semantic-ui-react';

const UsersTable = ({ clientes, onEdit, onAdd, onDelete }) => {
  return (
    <div className="clientes-clientes-table" data-test="clientes-table">
      <Table>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell>Id</Table.HeaderCell>
            <Table.HeaderCell>Nombre</Table.HeaderCell>
            <Table.HeaderCell>
              <Button className="clientes-clientes-table-onadd" onClick={() => onAdd()}>
                <Icon name="add" />
              </Button>
            </Table.HeaderCell>
          </Table.Row>
        </Table.Header>
        <Table.Body>
          {clientes.map(row => {
            return (
              <Table.Row key={row.id}>
                <Table.Cell component="th" scope="row">
                  {row.id}
                </Table.Cell>
                <Table.Cell>{row.nombre}</Table.Cell>
                <Table.Cell>
                  <Button
                    className="clientes-clientes-table-onedit"
                    icon="pencil"
                    aria-label="Create"
                    onClick={() => onEdit(row.id)}
                  />
                  <Button
                    className="clientes-clientes-table-ondelete"
                    icon="trash"
                    aria-label="Create"
                    onClick={() => onDelete(row.id)}
                  />
                </Table.Cell>
              </Table.Row>
            );
          })}
        </Table.Body>
      </Table>
    </div>
  );
};

UsersTable.propTypes = {
  clientes: PropTypes.array.isRequired,
  onEdit: PropTypes.func.isRequired,
  onAdd: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default UsersTable;
