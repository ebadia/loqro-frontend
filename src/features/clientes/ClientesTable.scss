@import '../../styles/mixins';

.clientes-clientes-table {

}
