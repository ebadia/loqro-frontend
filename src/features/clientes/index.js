export { default as DefaultPage } from './DefaultPage';
export { default as ClientesListPage } from './ClientesListPage';
export { default as ClienteDetailsPage } from './ClienteDetailsPage';
export { default as ClientesTable } from './ClientesTable';
export { default as ClienteForm } from './ClienteForm';
