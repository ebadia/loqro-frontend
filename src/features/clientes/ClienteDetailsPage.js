import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from './redux/actions';
import { Header, Button, Icon } from 'semantic-ui-react';
import { Form } from 'react-final-form';

import ClienteForm from './ClienteForm';

export class ClienteDetailsPage extends Component {
  static propTypes = {
    clientes: PropTypes.object.isRequired,
    actions: PropTypes.object.isRequired,
  };

  async componentDidMount() {
    const { fetchClienteDetails } = this.props.actions;
    const { match } = this.props;

    if (match.params.id) {
      await fetchClienteDetails(match.params.id);
    }
  }

  // handleSubmit = async (values, { props = this.props, setSubmitting }) => {
  handleSubmit = async values => {
    const { saveClienteDetails, addCliente } = this.props.actions;
    const { currentCliente } = this.props.clientes;
    if (currentCliente) {
      await saveClienteDetails(values);
      this.handleBack();
    } else {
      await addCliente(values);
      this.handleBack();
    }
  };

  handleBack() {
    const { history } = this.props;
    const { resetCurrentCliente } = this.props.actions;
    resetCurrentCliente();
    history.goBack();
  }

  validate = values => {
    const errors = {};
    if (!values.nombre) {
      errors.nombre = true;
    }
    return errors;
  };

  render() {
    const { currentCliente, fetchClienteDetailsPending } = this.props.clientes;

    return (
      <div className="clientes-cliente-details-page">
        <Header className="clientes-cliente-details-page-header" as="h1" color="teal">
          <Button className="clientes-cliente-details-page-back" onClick={() => this.handleBack()}>
            <Icon name="arrow left" />
          </Button>
          Cliente
        </Header>
        {!fetchClienteDetailsPending ? (
          <Form
            className="clientes-cliente-details-page-form"
            initialValues={currentCliente}
            onSubmit={this.handleSubmit}
            validate={this.validate}
            render={formprops => <ClienteForm {...formprops} />}
          />
        ) : (
          <div>Cargando...</div>
        )}
      </div>
    );
  }
}

/* istanbul ignore next */
function mapStateToProps(state) {
  return {
    clientes: state.clientes,
  };
}

/* istanbul ignore next */
function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({ ...actions }, dispatch),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ClienteDetailsPage);
