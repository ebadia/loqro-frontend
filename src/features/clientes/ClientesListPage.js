import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as actions from './redux/actions';
import { Header, Confirm } from 'semantic-ui-react';
import ClientesTable from './ClientesTable';

export class ClientesListPage extends Component {
  static propTypes = {
    clientes: PropTypes.object.isRequired,
    actions: PropTypes.object.isRequired,
  };

  state = {
    open: false,
    id: null,
  };

  componentDidMount() {
    const { fetchClientesList } = this.props.actions;
    fetchClientesList();
  }

  handleEdit(id) {
    const { history } = this.props;
    history.push(`/main/clientes/details/${id}`);
  }

  handleDelete = id => {
    this.setState({ open: true, id });
  };

  handleConfirm = async () => {
    const { remove } = this.props.actions;
    const { id } = this.state;
    await remove(id);
    this.setState({ open: false, id: null });
  };

  handleCancel = () => {
    this.setState({ open: false, id: null });
  };

  handleNewClient = () => {
    const { history } = this.props;
    const { resetCurrentCliente } = this.props.actions;
    resetCurrentCliente();
    history.push(`/main/clientes/details`);
  };

  render() {
    const { clientesList } = this.props.clientes;
    const { fetchCientesListError } = this.props.actions;
    const { open } = this.state;

    return (
      <React.Fragment>
        <div className="clientes-clientes-list-page-header">
          <Header className="clientes-clientes-list-page" as="h1" color="teal">
            Lista de clientes
          </Header>
          {!fetchCientesListError ? (
            <ClientesTable
              className="clientes-clientes-list-page-table"
              clientes={clientesList}
              onEdit={id => this.handleEdit(id)}
              onAdd={() => this.handleNewClient()}
              onDelete={id => this.handleDelete(id)}
            />
          ) : (
            <div>Error</div>
          )}
          <Confirm
            open={open}
            cancelButton="No"
            confirmButton="Si, adelante"
            content="Â¿Borrar el cliente?"
            onCancel={() => this.handleCancel()}
            onConfirm={() => this.handleConfirm()}
          />
        </div>
      </React.Fragment>
    );
  }
}

/* istanbul ignore next */
function mapStateToProps(state) {
  return {
    clientes: state.clientes,
  };
}

/* istanbul ignore next */
function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators({ ...actions }, dispatch),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ClientesListPage);
