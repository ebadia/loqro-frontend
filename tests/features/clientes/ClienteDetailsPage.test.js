import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { ClienteDetailsPage } from '../../../src/features/clientes/ClienteDetailsPage';

let wrapper;
const props = {
  clientes: { id: 1 },
  history: {
    goBack: jest.fn,
  },
  actions: {
    resetCurrentCliente: jest.fn,
    saveClienteDetails: jest.fn,
    addCliente: jest.fn,
  },
};

describe('<ClienteDetailsPage />', () => {
  beforeEach(() => {
    wrapper = shallow(<ClienteDetailsPage {...props} />);
  });

  it('renders correct', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders node with correct class name', () => {
    // expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('.clientes-cliente-details-page').length).toBe(1);
    expect(wrapper.find('.clientes-cliente-details-page-header').length).toBe(1);
    expect(wrapper.find('.clientes-cliente-details-page-back').length).toBe(1);
  });

  it('performs a back button click', () => {
    const onBackClick = sinon.spy(ClienteDetailsPage.prototype, 'handleBack');

    wrapper.find('.clientes-cliente-details-page-back').simulate('click');
    expect(onBackClick.calledOnce).toBeTruthy();
  });
});
