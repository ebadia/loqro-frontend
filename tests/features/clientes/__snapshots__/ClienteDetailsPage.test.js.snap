// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClienteDetailsPage /> renders correct 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ClienteDetailsPage
    actions={
        Object {
            "addCliente": [Function],
            "resetCurrentCliente": [Function],
            "saveClienteDetails": [Function],
          }
    }
    clientes={
        Object {
            "id": 1,
          }
    }
    history={
        Object {
            "goBack": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          as="h1"
          className="clientes-cliente-details-page-header"
          color="teal"
>
          <Button
                    as="button"
                    className="clientes-cliente-details-page-back"
                    onClick={[Function]}
                    role="button"
          >
                    <Icon
                              as="i"
                              name="arrow left"
                    />
          </Button>
          Cliente
</Header>,
        <ReactFinalForm
          className="clientes-cliente-details-page-form"
          initialValues={undefined}
          onSubmit={[Function]}
          render={[Function]}
          validate={[Function]}
/>,
      ],
      "className": "clientes-cliente-details-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "children": Array [
            <Button
              as="button"
              className="clientes-cliente-details-page-back"
              onClick={[Function]}
              role="button"
>
              <Icon
                            as="i"
                            name="arrow left"
              />
</Button>,
            "Cliente",
          ],
          "className": "clientes-cliente-details-page-header",
          "color": "teal",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "button",
              "children": <Icon
                as="i"
                name="arrow left"
/>,
              "className": "clientes-cliente-details-page-back",
              "onClick": [Function],
              "role": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "name": "arrow left",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "Cliente",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "clientes-cliente-details-page-form",
          "initialValues": undefined,
          "onSubmit": [Function],
          "render": [Function],
          "validate": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            as="h1"
            className="clientes-cliente-details-page-header"
            color="teal"
>
            <Button
                        as="button"
                        className="clientes-cliente-details-page-back"
                        onClick={[Function]}
                        role="button"
            >
                        <Icon
                                    as="i"
                                    name="arrow left"
                        />
            </Button>
            Cliente
</Header>,
          <ReactFinalForm
            className="clientes-cliente-details-page-form"
            initialValues={undefined}
            onSubmit={[Function]}
            render={[Function]}
            validate={[Function]}
/>,
        ],
        "className": "clientes-cliente-details-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "children": Array [
              <Button
                as="button"
                className="clientes-cliente-details-page-back"
                onClick={[Function]}
                role="button"
>
                <Icon
                                as="i"
                                name="arrow left"
                />
</Button>,
              "Cliente",
            ],
            "className": "clientes-cliente-details-page-header",
            "color": "teal",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "button",
                "children": <Icon
                  as="i"
                  name="arrow left"
/>,
                "className": "clientes-cliente-details-page-back",
                "onClick": [Function],
                "role": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "i",
                  "name": "arrow left",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "Cliente",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "clientes-cliente-details-page-form",
            "initialValues": undefined,
            "onSubmit": [Function],
            "render": [Function],
            "validate": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
